### DISCUSSION TITLE
Combining `its` functions

### DISCUSSION THREAD

**Discussion by pburrows** [winkjs/wink-nlp#44] [Help on coding] [answered]
Is there a built-in way to combine the output of multiple its? I want to do something like:

```
doc.tokens().out([its.normal, its.pos], as.freqTable)
```

To produce a list of unique tokens in descending order by frequency, grouped by token and part of speech (so if "google" is used both as a noun and a verb it would have two entries in the output)

I just want to make sure I'm not missing a built-in method. 

In the meantime, I'll explore writing my own `its` function to do this.

**Comment by pburrows** [comment, marked as answer]
Here's my solution to the problem, using as much of the built-in winkjs pieces that made sense.  Again, am curious if there is a better built-in way to handle this, even if it doesn't produce the exact same output format (but includes all the data).

Note for future people wanting to use this: these methods are private members of an ES6 class, hence the `#` at the start of the methods. To turn these into more "normal" javascript functions, just replace the `#` with the word `function `:

`#getWordsByPos` takes a parsed document that is the result of a `nlp.readDoc(xxx)` call, and an array of [Part-Of-Speech tags](https://winkjs.org/wink-nlp/part-of-speech.html) that you want to filter to. You could easily remove the filter step.

```
    #getWordsByPos(doc, posArray) {
        const taggedWords = []
        const items = doc.tokens();
        const filtered = items.filter(t => !t.parentEntity() && (posArray.indexOf(t.out(its.pos)) > -1));
        filtered.each((f) => {
            const pos = f.out(its.pos)
            if (posArray.indexOf(pos) > -1) {
                taggedWords.push({word: f.out(), pos: pos})
            }
        })

        const taggedWordsByFreq = this.#toFreqTable(taggedWords)

        return taggedWordsByFreq
    }

    #toFreqTable(taggedWords) {
        const bow = as.bow(taggedWords.map(p => `${p.word}__pos__${p.pos}`));
        const keys = Object.keys(bow);
        const length = keys.length;
        const result = new Array(length)
        for (let i = 0; i < length; i++) {
            result[i] = [keys[i], bow[keys[i]]]
        }
        result.sort(sort4FT)
        return result.map(p => {
            const parts = p[0].split('__pos__')
            return {
                word: parts[0],
                pos: parts[1],
                freq: p[1]
            }
        })
    }

```

**Comment by sanjayaksaxena** [comment]
Many thanks @pburrows for highlighting a scenario and also the possible solution. Your observation is correct, `.out()` does not support  `([its.normal, its.pos], as.freqTable)` like constructs.
